.github/workflows/main.yaml

```diff
@@ -0,0 +1,96 @@
+name: Provision Terraform Resources
+
+on:
+  push:
+    branches: [ INTERN-679-PIPELINE ]
+  pull_request:
+    branches: [ main ]
+
+
+env:
+  terraformVersion: 1.8.2
+  awsAccessKeyId: ${{ secrets.AWS_ACCESS_KEY_ID }}
+  awsSecretAccessKey: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+  awsSessionToken: ${{ secrets.AWS_SESSION_TOKEN }}
+  awsRegion: us-east-1
+  backendS3Bucket: 8586-terraform-state
+  backendS3Region: us-east-1
+  backendS3AccountId: ${{ secrets.ACCOUNT_ID }}
+  backendS3StateFile: anish_lab_pipeline2.tfstate
+
+  tfPrerequisiteResource: module.s3_bucket
+
+jobs:
+  create_s3:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup Terraform
+        uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_version: ${{ env.terraformVersion }}
+          terraform_wrapper: false
+
+      - name: configure aws credentials
+        uses: aws-actions/configure-aws-credentials@v3
+        with:
+          aws-access-key-id: ${{ env.awsAccessKeyId }}
+          aws-secret-access-key: ${{ env.awsSecretAccessKey }}
+          aws-session-token: ${{ env.awsSessionToken }}
+          aws-region: ${{ env.awsRegion }}
+
+      - name: Terraform Initialize
+        id: init
+        run: cd terraform && terraform init -backend-config="bucket=${{ env.backendS3Bucket }}" -backend-config="key=${{ env.backendS3AccountId }}/${{ env.backendS3StateFile }}" -backend-config="region=${{ env.backendS3Region }}"
+
```

Comment:
Also add the plan phase for terraform.
If terrform plan succeeds then only move to terraform apply.

Reply:
Added plan